src/testing/fast_array_utils/__init__.py

```diff
@@ -37,48 +40,224 @@ def __call__(  # noqa: D102
+            case "cupyx.scipy.sparse", ("csr_matrix" | "csc_matrix") as cls_name, None:
+                import cupyx.scipy.sparse as cu_sparse
+
+                return getattr(cu_sparse, cls_name)  # type: ignore[no-any-return]
+            case "dask.array", cls_name, _:
+                if TYPE_CHECKING:
+                    from dask.array.core import Array as DaskArray
+                else:
+                    from dask.array import Array as DaskArray
+
+                return DaskArray
+            case "h5py", "Dataset", _:
+                import h5py
+
+                return h5py.Dataset  # type: ignore[no-any-return]
+            case "zarr", "Array", _:
+                import zarr
+
+                return zarr.Array
+            case _:
+                msg = f"Unknown array class: {self}"
+                raise ValueError(msg)
+
+    def random(
+        self,
+        shape: tuple[int, int],
+        *,
+        dtype: _DTypeLikeFloat32 | _DTypeLikeFloat64 | None,
+        gen: np.random.Generator | None = None,
+    ) -> Array:
+        """Create a random array."""
+        gen = np.random.default_rng(gen)
+
+        match self.mod, self.name, self.inner:
+            case "numpy", "ndarray", None:
+                return gen.random(shape, dtype=dtype or np.float64)
+            case "scipy.sparse", (
+                "csr_array" | "csc_array" | "csr_matrix" | "csc_matrix"
+            ) as cls_name, None:
+                fmt, container = cls_name.split("_")
+                return random_mat(shape, format=fmt, container=container, dtype=dtype)  # type: ignore[arg-type]
+            case "cupy", "ndarray", None:
+                raise NotImplementedError
+            case "cupyx.scipy.sparse", ("csr_matrix" | "csc_matrix") as cls_name, None:
+                raise NotImplementedError
+            case "dask.array", cls_name, _:
+                raise NotImplementedError
+            case "h5py", "Dataset", _:
+                raise NotImplementedError
+            case "zarr", "Array", _:
+                raise NotImplementedError
+            case _:
+                msg = f"Unknown array class: {self}"
+                raise ValueError(msg)
+
+    def __call__(self, x: ArrayLike, /, *, dtype: DTypeLike | None = None) -> Array:
+        """Convert to this array type."""
+        from fast_array_utils import types
+
+        fn: ToArray
+        if self.cls is np.ndarray:
+            fn = np.asarray  # type: ignore[assignment]
+        elif self.cls is types.DaskArray:
+            if self.inner is None:
+                msg = "Cannot convert to dask array without inner array type"
+                raise AssertionError(msg)
+            fn = self.to_dask_array
+        elif self.cls is types.H5Dataset:
+            fn = self.to_h5py_dataset
+        elif self.cls is types.ZarrArray:
+            fn = self.to_zarr_array
+        elif self.cls is types.CupyArray:
```

Comment:
what about cupy sparse?

Reply:
I don’t test cupy yet, so that’ll be fun. For a few classes I rely on the else branch:

```python
else:
    fn = self.cls
```

e.g. `csr_array([[1,2,3], [4,5,6]])` works, no need for extra handling.